+neuron/Session.m

```diff
@@ -10,15 +11,17 @@
         null            % clib.type.nullptr
     end
 
-    methods
-        function self = Neuron()
+    methods(Hidden)
```

Comment:
With a Hidden constructor, it is still callable if you know it exists (which is needed for neuron.launch() to call it) and is not a Singleton pattern. So if neuron.launch is called twice, it will return a different object, that will call clib.neuron.initialize.

To have a singleton, the constructor needs to be private, with a static method to get an instance, a persistent variable in the static method to store the instance, and a mlock on the file (and munlock somewhere) to prevent the persistent variable from being accidentally cleared. Oh, and a destructor to clear the persistent variable.

This has it using two classes, not sure that is actually needed: https://nl.mathworks.com/matlabcentral/fileexchange/24911-design-pattern-singleton-creational
And, that does not have the mlock, so a clear all or a clear classes would clear the instance. That can be ok, if you have a destructor being called in that case that closes the neuron.